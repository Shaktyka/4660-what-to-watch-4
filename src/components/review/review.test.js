import React from 'react';
import renderer from 'react-test-renderer';

import Review from './review.jsx';

const reviewData = {
  id: 1,
  authorId: 22,
  text: `Discerning travellers`,
  authorName: `Kate Muir`,
  date: `2016-12-24`,
  rating: `8,9`
};

describe(`Review rendering`, () => {

  it(`Review renders correctly`, () => {
    const {text, authorName, date, rating} = reviewData;

    const tree = renderer
      .create(
          <Review
            text={text}
            author={authorName}
            date={date}
            rating={rating}
          />, {
            createNodeMock: () => {
              return {};
            }
          }
      )
      .toJSON();

    expect(tree).toMatchSnapshot();
  });

});
