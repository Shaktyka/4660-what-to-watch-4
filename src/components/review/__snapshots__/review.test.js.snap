// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Review rendering Review renders correctly 1`] = `
<div
  className="review"
  id={1}
>
  <blockquote
    className="review__quote"
  >
    <p
      className="review__text"
    >
      Discerning travellers
    </p>
    <footer
      className="review__details"
    >
      <cite
        className="review__author"
        data-authorid={22}
      >
        Kate Muir
      </cite>
      <time
        className="review__date"
        dateTime={"2016-12-23T19:00:00.000Z"}
      >
        December 24, 2016
      </time>
    </footer>
  </blockquote>
  <div
    className="review__rating"
  >
    8.9
  </div>
</div>
`;
